### PR TITLE
various fixes + authentication improvements

### PR DESCRIPTION
**Pull request by EvanBldy**
**Problem**
- In edits page department preselection is not working.
- When we create a new user the "invite" button can be clicked if user does not exist.
- In People page when a user is created the new user is not in the list if we don't reload the page.
- src/components/modals/EditPersonDepartmentsModal.vue file is unused.
- Admin can't change passwords of other users.
- In People page when we have unactive users and we scroll the column name is not staying at the left as for active users.
- When LDAP is activated among other things we can't : create new users / edit existing users (not generated from LDAP).
- We need a security when a lot of logins fails for the same user.

**Solution**
- Fix department preselection in edits page.
- When creating a new user "invite" button is now disabled if user does not exist.
- Properly reload users in People page if a new user is created.
- Remove unused file : src/components/modals/EditPersonDepartmentsModal.vue.
- Admin can now change password of other users (only can't change passwords of other admins).
- Fix scrolling issue for unactive users in People page.
- Allow to create and edit users not from ldap (and change passwords).
- When login fails multiple times for same user, force to wait before an other attempt.

This PR is related to this PR from Zou : https://github.com/cgwire/zou/pull/533

### CODE REVIEW

File: src/locales/en.js [hunk 1]
```diff
@@ -296,6 +296,7 @@ export default {
     forgot_password: 'Forgot password?',
     login: 'Log in',
     login_failed: 'Log in failed, please verify your credentials',
+    too_much_login_failed_attemps: 'Too much login failed attemps, please wait one minute',
```
Comment (jhoolmans) [inline src/locales/en.js:299]: Small correction:
'Too many failed login attempts, please wait one minute'

Comment (EvanBldy) [reply src/locales/en.js:299]: Yeah thanks :)

File: src/components/modals/ChangePasswordModal.vue [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import { modalMixin } from '@/components/modals/base_modal'
+
+import TextField from '@/components/widgets/TextField'
+
+import auth from '@/lib/auth'
+
+export default {
+  name: 'change-password-modal',
+  mixins: [modalMixin],
+  props: [
```
Comment (frankrousseau) [inline src/components/modals/ChangePasswordModal.vue:83]: Please use descriptors for props like in the other components.

File: src/components/modals/ChangePasswordModal.vue [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  },
+
+  methods: {
+    ...mapActions([
+      'changePasswordPerson'
+    ]),
+
+    confirmClicked () {
+      if (auth.isPasswordValid(this.form.password, this.form.password2)) {
+        this.changePasswordPerson({
```
Comment (frankrousseau) [inline src/components/modals/ChangePasswordModal.vue:116]: Can you convert this to a promise?

File: src/components/pages/Login.vue [hunk 4]
```diff
@@ -12,6 +12,7 @@
               <input
                 class="input is-medium email"
                 type="text"
+                v-model="email"
```
Comment (frankrousseau) [inline src/components/pages/Login.vue:15]: Always put  `v-model` at the bottom for consistency reasons.

File: src/locales/en.js [hunk 5]
```diff
@@ -845,7 +847,7 @@ export default {
       button: 'Change password',
       error: 'An error occurred while changing password. Please verify your current password.',
       success: 'Your password was successfully changed!',
-      unvalid: 'Your new password confirmation doesn\'t match or your password is too short (7 chars, at least, is expected).'
+      unvalid: 'Your new password confirmation doesn\'t match or your password is too short (8 chars, at least, is expected).'
```
Comment (frankrousseau) [inline src/locales/en.js:850]: Don't add translations, they are managed via POEditor.

File: src/components/modals/ChangePasswordModal.vue (hunk 2 above)
Comment (EvanBldy) [reply src/components/modals/ChangePasswordModal.vue:83]: It's fixed

File: src/components/pages/Login.vue (hunk 4 above)
Comment (EvanBldy) [reply src/components/pages/Login.vue:15]: It's fixed

File: src/components/modals/ChangePasswordModal.vue (hunk 3 above)
Comment (EvanBldy) [reply src/components/modals/ChangePasswordModal.vue:116]: It's fixed

**Pull request merged**